petric.py

```diff
@@ -66,20 +69,30 @@ def __call__(self, algo: Algorithm):
             self.tb.add_scalar("normalised_change", normalised_change, algo.iteration)
         self.x_prev = algo.x.clone()
         self.tb.add_image("transverse",
-                          clip(algo.x.as_array()[self.transverse_slice:self.transverse_slice + 1] / self.vmax, 0, 1),
+                          np.clip(algo.x.as_array()[self.transverse_slice:self.transverse_slice + 1] / self.vmax, 0, 1),
                           algo.iteration)
-        self.tb.add_image("coronal", clip(algo.x.as_array()[None, :, self.coronal_slice] / self.vmax, 0, 1),
+        self.tb.add_image("coronal", np.clip(algo.x.as_array()[None, :, self.coronal_slice] / self.vmax, 0, 1),
                           algo.iteration)
 
 
 class MetricsWithTimeout(cbks.Callback):
     """Stops the algorithm after `seconds`"""
-    def __init__(self, seconds=300, outdir=OUTDIR, transverse_slice=None, coronal_slice=None, verbose=1):
+    def __init__(self, seconds=300, outdir=OUTDIR, transverse_slice=None, coronal_slice=None, ground_truth=None,
+                 verbose=1):
         super().__init__(verbose)
         self.callbacks = [
             cbks.ProgressCallback(),
             SaveIters(outdir=outdir),
-            TensorBoard(logdir=outdir, transverse_slice=transverse_slice, coronal_slice=coronal_slice)]
+            (tb_cbk := TensorBoard(logdir=outdir, transverse_slice=transverse_slice, coronal_slice=coronal_slice))]
+
+        if ground_truth:
+            roi_image_dict = {f'S{i}': STIR.ImageData(f'S{i}.hv') for i in range(1, 8)}
+            self.callbacks.append(
+                ImageQualityCallback(
```

Comment:
Add a comment that this is for testing only, and that these are not part of the metrics to be used for the challenge